ibm/service/kubernetes/resource_ibm_container_vpc_worker.go

```diff
@@ -0,0 +1,577 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package kubernetes
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/bluemix-go/bmxerror"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+// Mutex to make resource creation sequential.
+var resourceIBMContainerVpcWorkerCreateMutex sync.Mutex
+var commonVarMutex sync.Mutex
+
+// Status of worker replace
+var workerReplaceStatus bool = false
+var replaceInProgress bool = false
+
+// Variable to identify the first run
+var initRun int = 1
+
+func ResourceIBMContainerVpcWorker() *schema.Resource {
+
+	return &schema.Resource{
+		Create:   resourceIBMContainerVpcWorkerCreate,
+		Read:     resourceIBMContainerVpcWorkerRead,
+		Delete:   resourceIBMContainerVpcWorkerDelete,
+		Exists:   resourceIBMContainerVpcWorkerExists,
+		Importer: &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Cluster name",
+			},
+
+			"replace_worker": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "Worker name/id that needs to be replaced",
+			},
+
+			"resource_group_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "ID of the resource group.",
+			},
+
+			"kube_config_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     nil,
+				Description: "Path of downloaded cluster config",
+			},
+
+			"check_ptx_status": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     true,
+				Description: "Check portworx status after worker replace",
+			},
+
+			"ptx_timeout": {
```

Comment:
is forcenew : true required here? if we change value it will destroy or recreate the resource

Reply:
Getting this error, if we dont have forcenew specified
```
1 error occurred:
│ 	* resource ibm_container_vpc_worker: No Update defined, must set ForceNew on: []string{"ptx_timeout"}
│ 
```